### ISSUE TITLE
[Feature Request] Provide ESM bundle

### ISSUE THREAD

**Issue by himself65** [facebook/stylex#205]
**The problem**
```shell
4:18:25 PM [vite] Error when evaluating SSR module /@fs/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:
|- ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15)

4:18:25 PM [vite] Error when evaluating SSR module /src/components/Banner.tsx: failed to import "/@fs/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js"
|- ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15)

4:18:25 PM [vite] Error when evaluating SSR module /src/components/App.tsx: failed to import "/src/components/Banner.tsx"
|- ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15)

ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15)
ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15) {
  digest: ''
}
Cannot render RSC ReferenceError: exports is not defined
    at eval (/Users/himself65/Code/waku/node_modules/.pnpm/@stylexjs+stylex@0.3.0/node_modules/@stylexjs/stylex/lib/stylex.js:5:23)
    at instantiateModule (file:///Users/himself65/Code/waku/node_modules/.pnpm/vite@5.0.9_@types+node@20.10.4/node_modules/vite/dist/node/chunks/dep-Pluk1iaB.js:50504:15) {
  digest: ''
}
```

**How to reproduce**

Steps to reproduce:
1.
2.
3.

**Expected behavior**

**Environment (include versions). Did this work in previous versions?**

**Comment by nmn** [comment]
We will work on this. In the meantime, can you verify that you see the same issues with a different package manager?

Also, would you be willing to share an example app for this (perhaps as a StackBlitz?) so that we can verify our solution.

**Comment by nmn** [comment]
Fixed in v0.5.0
